test/validation/api/scheduler/scheduler.c

```diff
@@ -493,6 +495,133 @@ static void scheduler_test_queue_size(void)
+{
+	odp_schedule_config_t default_config;
+	odp_pool_t pool;
+	odp_pool_capability_t pool_capa;
+	odp_pool_param_t pool_param;
+	odp_schedule_capability_t sched_capa;
+	odp_queue_param_t queue_param;
+	odp_event_t ev;
+	uint32_t i, j, k, num_bufs, events_per_queue, num_queues;
+	uint32_t queue_size = 2048;
+	uint32_t num_enq = 0;
+	uint32_t num = 0;
+	int ret;
+	odp_schedule_sync_t sync[] = {ODP_SCHED_SYNC_PARALLEL,
+				      ODP_SCHED_SYNC_ATOMIC,
+				      ODP_SCHED_SYNC_ORDERED};
+
+	CU_ASSERT_FATAL(!odp_schedule_capability(&sched_capa));
+	if (sched_capa.max_queue_size && queue_size > sched_capa.max_queue_size)
+		queue_size = sched_capa.max_queue_size;
+
+	/* Scheduler has been already configured. Use default config as queue
+	 * size and queue count. */
+	odp_schedule_config_init(&default_config);
+	if (default_config.queue_size)
+		queue_size = default_config.queue_size;
+	num_queues = default_config.num_queues;
+
+	/* Queues reserved by create_queues() in suite init */
+	num_queues -= globals->num_queues;
+
+	odp_queue_t queue[num_queues];
+
+	CU_ASSERT_FATAL(!odp_pool_capability(&pool_capa));
+	num_bufs = num_queues * queue_size;
+	if (pool_capa.buf.max_num && num_bufs > pool_capa.buf.max_num)
+		num_bufs = pool_capa.buf.max_num;
+	if (num_bufs > MAX_POOL_SIZE)
+		num_bufs = MAX_POOL_SIZE;
+	events_per_queue = num_bufs / num_queues;
+
+	/* Make sure there is at least one event for each queue */
+	while (events_per_queue == 0) {
+		num_queues--;
+		events_per_queue = num_bufs / num_queues;
+	}
+
+	odp_pool_param_init(&pool_param);
+	pool_param.buf.size  = 100;
+	pool_param.buf.align = 0;
+	pool_param.buf.num   = num_bufs;
+	pool_param.type      = ODP_POOL_BUFFER;
+
+	pool = odp_pool_create("test_full_queues", &pool_param);
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	/* Ensure that scheduler is empty */
+	drain_queues();
+
+	/* Run test for each scheduler synchronization type */
+	for (i = 0; i < 3; i++) {
+		/* Create and fill all queues */
+		for (j = 0; j < num_queues; j++) {
+			odp_queue_param_init(&queue_param);
+			queue_param.type = ODP_QUEUE_TYPE_SCHED;
+			queue_param.sched.prio  = odp_schedule_default_prio();
+			queue_param.sched.sync  = sync[i];
+			queue_param.sched.group = ODP_SCHED_GROUP_ALL;
+			queue_param.size = events_per_queue;
+
+			queue[j] = odp_queue_create("test_full_queues",
+						    &queue_param);
+			CU_ASSERT_FATAL(queue[j] != ODP_QUEUE_INVALID);
+
+			for (k = 0; k < events_per_queue; k++) {
+				odp_buffer_t buf = odp_buffer_alloc(pool);
+
+				CU_ASSERT(buf != ODP_BUFFER_INVALID);
+				if (buf == ODP_BUFFER_INVALID)
+					continue;
+
+				ev = odp_buffer_to_event(buf);
+				ret = odp_queue_enq(queue[j], ev);
+				CU_ASSERT(ret == 0);
+				if (ret) {
+					odp_event_free(ev);
+					continue;
+				}
+				num_enq++;
+			}
+		}
+		/* Run normal scheduling rounds */
+		for (j = 0; j < num_bufs; j++) {
+			odp_queue_t src_queue;
+
+			ev = odp_schedule(&src_queue, ODP_SCHED_NO_WAIT);
```

Comment:
HW may have small delay from enqueue to schedule. May be this schedule call could also have a small wait time to ensure that test won't fail due to HW delays.

Reply:
For that reason there is no CU_ASSERT check for this schedule call. The clean-up loop, which uses wait time, checks that all enqueued events are received.